substrate/zombienet/zombienet-sdk/tests/zombie_ci/block_building_warp_sync.rs

```diff
@@ -0,0 +1,311 @@
+		network
+			.get_node(validator)?
+			.wait_metric_with_timeout(
+				"node_roles",
+				|role| role == VALIDATOR_ROLE_VALUE,
+				ROLE_TIMEOUT_SECS,
+			)
+			.await?;
+	}
+
+	for follower in FULLNODES {
+		network
+			.get_node(follower)?
+			.wait_metric_with_timeout(
+				"node_roles",
+				|role| role == FULLNODE_ROLE_VALUE,
+				ROLE_TIMEOUT_SECS,
+			)
+			.await?;
+	}
+
+	Ok(())
+}
+
+async fn check_peers(
+	network: &zombienet_sdk::Network<zombienet_sdk::LocalFileSystem>,
+) -> Result<()> {
+	for &node_name in VALIDATORS.iter().chain(FULLNODES.iter()) {
+		network
+			.get_node(node_name)?
+			.wait_metric_with_timeout(
+				"substrate_sub_libp2p_peers_count",
+				|peers| peers >= PEERS_THRESHOLD,
+				PEER_TIMEOUT_SECS,
+			)
+			.await?;
+	}
+
+	Ok(())
+}
+
+async fn verify_bootstrap_height(
+	network: &zombienet_sdk::Network<zombienet_sdk::LocalFileSystem>,
+	db_snapshot_height: f64,
+) -> Result<()> {
+	for &node_name in VALIDATORS.iter().chain(FULLNODES.iter()) {
+		network
+			.get_node(node_name)?
+			.wait_metric_with_timeout(
+				BEST_BLOCK_METRIC,
+				|height| height >= MIN_BOOTSTRAP_BLOCK,
+				BOOTSTRAP_TIMEOUT_SECS,
+			)
+			.await?;
+		network
+			.get_node(node_name)?
+			.wait_metric_with_timeout(
+				BEST_BLOCK_METRIC,
+				|height| height >= db_snapshot_height,
+				METRIC_TIMEOUT_SECS,
+			)
+			.await?;
+	}
+
+	Ok(())
+}
+
+async fn verify_new_blocks(
+	network: &zombienet_sdk::Network<zombienet_sdk::LocalFileSystem>,
+	db_snapshot_height: f64,
+) -> Result<()> {
+	for node_name in VALIDATORS {
+		network
+			.get_node(node_name)?
+			.wait_metric_with_timeout(
+				BEST_BLOCK_METRIC,
+				|height| height > db_snapshot_height,
+				NEW_BLOCK_TIMEOUT_SECS,
+			)
+			.await?;
+	}
+
+	let charlie = network.get_node("charlie")?;
+	charlie
+		.wait_metric_with_timeout(
+			BEST_BLOCK_METRIC,
+			|height| height > db_snapshot_height,
+			NEW_BLOCK_TIMEOUT_SECS,
+		)
+		.await?;
+
+	Ok(())
+}
+
+async fn verify_node_progress(node: &NetworkNode, db_snapshot_height: f64) -> Result<()> {
+	let node_name = node.name();
+
+	node.wait_metric_with_timeout(BEST_BLOCK_METRIC, |height| height >= 1.0, METRIC_TIMEOUT_SECS)
+		.await
+		.with_context(|| {
+			format!(
+				"{node_name} did not report BEST_BLOCK_METRIC >= 1 within {METRIC_TIMEOUT_SECS}s"
+			)
+		})?;
+	node.wait_metric_with_timeout(
+		BEST_BLOCK_METRIC,
+		|height| height >= db_snapshot_height,
+		METRIC_TIMEOUT_SECS,
+	)
+	.await
+	.with_context(|| {
+		format!(
+			"{node_name} did not catch up to snapshot height {db_snapshot_height} within {METRIC_TIMEOUT_SECS}s"
+		)
+	})?;
+	node.wait_metric_with_timeout(
+		BEST_BLOCK_METRIC,
+		|height| height > db_snapshot_height,
+		METRIC_TIMEOUT_SECS,
+	)
+	.await
+	.with_context(|| {
+		format!(
+			"{node_name} did not produce blocks beyond snapshot height {db_snapshot_height} within {METRIC_TIMEOUT_SECS}s"
+		)
+	})?;
+
+	Ok(())
+}
+
+async fn verify_node_logs(node: &NetworkNode) -> Result<()> {
```

Comment:
This is similar to `wait_for_warp_logs()` from the other test. 
Not sure, maybe it makes sense to make it common?